gpt4all-chat/src/chatmodel.h

```diff
@@ -125,75 +143,109 @@ class ChatModel : public QAbstractListModel
         ChatItem item;
         item.name = name;
         item.value = value;
-        beginInsertRows(QModelIndex(), m_chatItems.size(), m_chatItems.size());
+        m_mutex.lock();
+        const int count = m_chatItems.count();
+        m_mutex.unlock();
+        beginInsertRows(QModelIndex(), count, count);
+        m_mutex.lock();
         m_chatItems.append(item);
+        m_mutex.unlock();
```

Comment:
QList::append is not noexcept (and can't be due to alloc). Should use a scoped lock like above:
```suggestion
        {
            QMutexLocker locker(&m_mutex);
            m_chatItems.append(item);
        }
```